lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -23,15 +23,32 @@ const Sentry = require('../../lib/sentry.js');
 
 /**
  * @this {HTMLElement}
+ * @param {string} traceEventType
  */
 /* c8 ignore start */
-function getNodeDetailsData() {
+function getNodeDetailsData(traceEventType) {
   const elem = this.nodeType === document.ELEMENT_NODE ? this : this.parentElement; // eslint-disable-line no-undef
-  let traceElement;
-  if (elem) {
-    // @ts-expect-error - getNodeDetails put into scope via stringification
-    traceElement = {node: getNodeDetails(elem)};
+  if (!elem) return;
+
+  /** @type {import('../../lib/page-functions.js').NodeDetailsOptions=} */
+  let options;
+  if (elem.tagName === 'img' && traceEventType === 'largest-contentful-paint') {
```

Comment:
hmm.. i suppose this might as well run for any `image` element in trace-elements, or for that matter, any img element anywhere ...

Reply:
Yeah, seems like the problem was not specific to LCP. In that case, is it possible to move this into `getOuterHTMLSnippet`?

Reply:
+1 to inlining into `getOuterHTMLSnippet`